### ISSUE TITLE
Method override completion doesn't return the correct return type

### ISSUE THREAD

**Issue by StellaHuang95** [microsoft/pylance-release#5725]
## Environment data

-   Language Server version:2024.4.1

## Code Snippet

```python
# file1.py
class Parent:
    def method(self) -> str:
        pass
```
```python
# file2.py
from sys import version
from file1 import Parent

print(version)

class Child(Parent):
    def meth # <- trigger completion here
```

## Repro Steps

1. Type `meth` and accept completion

## Expected behavior
```python
class Child(Parent):
    def method(self) -> str:
        return super().method()
```

## Actual behavior

```python
class Child(Parent):
    def method(self) -> version:
        return super().method()
```

The return type is wrong.

**Comment by Viicos** [comment]
Similar, if not the same: https://github.com/microsoft/pylance-release/issues/5534

**Comment by StellaHuang95** [comment]
Looks like it only repros if the other symbol imported is the same type as the return type.

**Comment by luabud** [comment]
Can no longer repro this with v2024.5.1

**Comment by StellaHuang95** [comment]
Doesn't repro for me with the latest pylance version 2024.5.103. Closing it out.

**Comment by debonte** [comment]
This was a duplicate of https://github.com/microsoft/pylance-release/issues/5693 which was fixed in 2024.4.100.
